ocp_utilities/infra.py

```diff
@@ -81,6 +85,55 @@ def assert_pods_failed_or_pending(pods):
         )
 
 
+def assert_nodes_in_healthy_condition(nodes):
+    """
+    Validates nodes are in a healthy condition.
+    Nodes Ready condition is True and none of the following node conditions is True:
+        - DiskPressure
+        - MemoryPressure
+        - PIDPressure
+        - NetworkUnavailable
+
+    Args:
+         nodes(list): List of Node objects
+
+    Raises:
+        NodesNotHealthyConditionError: if any nodes DiskPressure
+        MemoryPressure, PIDPressure, NetworkUnavailable condition is True
+    """
+    LOGGER.info("Verify all nodes are in a healthy condition")
+
+    default_healthy_node_condition_type = dict(
+        DiskPressure=Node.Condition.Status.FALSE,
+        MemoryPressure=Node.Condition.Status.FALSE,
+        PIDPressure=Node.Condition.Status.FALSE,
+        NetworkUnavailable=Node.Condition.Status.FALSE,
+        Ready=Node.Condition.Status.TRUE,
+    )
+
+    unhealthy_nodes_with_conditions = {}
+    for node in nodes:
+
+        condition_type_list = [
+            condition.type
+            for condition in node.instance.status.conditions
+            if default_healthy_node_condition_type.get(condition.type)
+            != condition.status
+        ]
```

Comment:
`default_healthy_node_condition_type.get(condition.type)` will return None on new condition and the code will raise

you need to iterate over `for condition in node.instance.status.conditions`, if the key (condition.type) exists in `default_healthy_node_condition_type`, compare else continue